cdap-common/bin/functions.sh

```diff
@@ -668,12 +670,30 @@ cdap_start_java() {
     if [[ -n ${JAVA_LIBRARY_PATH} ]]; then
       __defines+=" -Djava.library.path=${JAVA_LIBRARY_PATH}"
     fi
-    __startup_checks=${CDAP_STARTUP_CHECKS:-$(cdap_get_conf "master.startup.checks.enabled" "${CDAP_CONF}"/cdap-site.xml true)}
+    # Check for HDP 2.2+ or IOP, otherwise do nothing and leave up to the user to configure
+    for __dist in hdp iop; do
+      if [[ $(which ${__dist}-select 2>/dev/null) ]]; then
+        local __auto_version=$(${__dist}-select status hadoop-client | '{print $3}')
+        # Check for version configured in OPTS
+        if [[ ${OPTS} =~ -D${__dist}.version ]]; then
+          local __conf_version=$(echo ${OPTS} | grep -oP "\-D${__dist}.version=\d+\.\d+\.\d+\.\d+-\d+" | cut -d= -f2)
+          if [[ ${__conf_version} != ${__auto_version} ]]; then
+            logecho "[WARN] ${__dist^^} version mismatch! Detected: ${__auto_version}, Configured: ${__conf_version}"
```

Comment:
There aren't really any concerns. However, this is the only use of a bash 4+ feature and it's only for output, so I'll replace it with a little awk.